### PR TITLE
Added support for delete multiple configs

### PR DESCRIPTION
**Pull request by akbhatta**
### Description
Added support for delete multiple configs
Added support for specifying ID while creating configurations
Added unit and integrations test for create and delete configs

Signed-off-by: @akbhatta

### Issues Resolved
[List any issues this PR will resolve]

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass, including unit test, integration test and doctest
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added
  - [x] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: notifications/src/main/kotlin/org/opensearch/commons/notifications/action/CreateNotificationConfigRequest.kt [hunk 1]
```diff
@@ -36,15 +36,18 @@ import org.opensearch.common.xcontent.ToXContentObject
...
 import org.opensearch.commons.notifications.model.NotificationConfig
+import org.opensearch.commons.utils.fieldIfNotNull
 import org.opensearch.commons.utils.logger
 import java.io.IOException
 
 /**
  * Action request for creating new configuration.
  */
 class CreateNotificationConfigRequest : ActionRequest, ToXContentObject {
+    val configId: String?
```
Comment (zhongnansu) [inline notifications/src/main/kotlin/org/opensearch/commons/notifications/action/CreateNotificationConfigRequest.kt:50]: Why do we need to specify id when creating Notification config? Why can't just use the default one that Opensearch generated for us?

**Review by zhongnansu — APPROVED** [review]
LGTM, left one comment

File: notifications/src/main/kotlin/org/opensearch/commons/notifications/action/CreateNotificationConfigRequest.kt (hunk 1 above)
Comment (akbhatta) [reply notifications/src/main/kotlin/org/opensearch/commons/notifications/action/CreateNotificationConfigRequest.kt:50]: it is an optional field. if the caller wants to use specific id for some reason, they can specify.

Comment (jayeshathila) [reply notifications/src/main/kotlin/org/opensearch/commons/notifications/action/CreateNotificationConfigRequest.kt:50]: Does it serve any particular use case where user wants to give a specific id. Reason being, in general REST API  `id` are generated by Backend service itself.

Comment (akbhatta) [reply notifications/src/main/kotlin/org/opensearch/commons/notifications/action/CreateNotificationConfigRequest.kt:50]: When other plugins/systems migrate to this new notification plugin, they may need to retain the IDs for easier migration. Dashboard UI is not expected to fill this field. This is mainly intended for tools and systems which are directly interacting with OpenSearch backend.

**Review by joshuali925 — APPROVED** [review]
LGTM

**Pull request merged**